sdks/python/apache_beam/runners/interactive/pipeline_instrument_test.py

```diff
@@ -273,6 +281,47 @@ def visit_transform(self, transform_node):
     p_origin.visit(v)
     assert_pipeline_equal(self, p_origin, p_copy)
 
+  def test_instrument_example_unbounded_pipeline_to_read_cache(self):
+    p_origin, init_pcoll, second_pcoll = self._example_pipeline(watch=True,
+                                                                bounded=False)
+    p_copy, _, _ = self._example_pipeline(watch=False, bounded=False)
+
+    # Mock as if cacheable PCollections are cached.
+    init_pcoll_cache_key = 'init_pcoll_' + str(
+        id(init_pcoll)) + '_' + str(id(init_pcoll.producer))
+    self._mock_write_cache(init_pcoll, init_pcoll_cache_key)
+    second_pcoll_cache_key = 'second_pcoll_' + str(
+        id(second_pcoll)) + '_' + str(id(second_pcoll.producer))
+    self._mock_write_cache(second_pcoll, second_pcoll_cache_key)
+    ie.current_env().cache_manager().exists = MagicMock(return_value=True)
+    instr.pin(p_copy)
```

Comment:
Is it intentional that we ignore the return PipelineInstrument object?

Outside of the scope of this PR: Should we change the method name "pin"? It's a confusing name. @KevinGG

Reply:
Yes, we can change it to `build_pipeline_instrument`.
Yes, it's intentional. The instrument process modifies the given pipeline which is already a copied runner pipeline.

Reply:
Thanks. Can we add a comment to the pin() method that clarifies?

Reply:
I can do it in my next PR. Sam can leave it be.